### PR TITLE
AHL on Schedule Veteran Modal

### LINKED ISSUE #8041: Geo matching | Schedule Veterans | Case Details Modal
As a hearing coordinator, I need a way to schedule veterans by alternate hearing location and ensure that the Veteran is going to their closest hearing location. 

### Acceptance Criteria
- [ ] Verify that when the user clicks "Schedule Veterans" from case details the schedule modal comes up
- [ ] Verify the following in the modal:
   - Title: Schedule Veteran Name
   - Veteran address underneath the title 
   - Regional Office: 
      Regional Office 
   - Suggested Hearing Location
      - Dropdown shows: Suggested hearing location (facility name) distance in miles
   - Hearing date: 
      Hearing day 
   - Hearing Time: 
      8:00 am and 12:30 p.m as radio buttons 
- [ ] Verify that there is a change link next to Regional Office and hearing day 
- [ ] Verify that when this change link is selected, a drop down appears
- [ ] Verify that changing the Regional Office, changes the suggested location and the available hearing day 

### Mocks

### PR DESCRIPTION
**Pull request by lomaxap**
Resolves #8041, #8307

- Adds hearing location to Schedule Veteran modal
- creates LegacyHearing when hearing is scheduled
- slight refactor of `slot_new_hearing`
- uses `slot_new_hearing` in ScheduleVeteranTask

### CODE REVIEW

File: app/repositories/hearing_repository.rb [hunk 1]
```diff
@@ -86,22 +86,26 @@ def to_hash(hearing)
...
+        create_child_video_hearing(
+          parent_record_id, hearing_datetime, appeal, hearing_location_attrs: hearing_location_attrs
+        )
       end
     end
 
-    def create_child_co_hearing(hearing_date_str, appeal)
+    def create_child_co_hearing(hearing_date_str, appeal, hearing_location_attrs: nil)
       hearing_day = HearingDay.find_by(request_type: "C", scheduled_for: hearing_date_str.to_date)
       fail LockedHearingDay, message: "Locked hearing day" if hearing_day.lock
```
Comment (lomaxap) [inline app/repositories/hearing_repository.rb]: should this method also make a central hearing for AMA vets nears DC?

File: client/app/queue/components/AssignHearingModal.jsx [hunk 2]
```diff
@@ -101,29 +141,30 @@ class AssignHearingModal extends React.PureComponent<Props, LocalState> {
...
+    } = this.props;
 
-    if (this.props.openHearing) {
-      return false;
-    }
+    const invalid = {
+      day: selectedHearingDay ? false : 'Please select a hearing day',
+      regionalOffice: selectedRegionalOffice ? false : 'Please select a regional office',
+      time: selectedHearingTime ? false : 'Please pick a hearing time'
+      // location: selectedHearingLocation ? false : 'Please select a hearing location'
```
Comment (lomaxap) [inline client/app/queue/components/AssignHearingModal.jsx]: I think I want to make this not mandatory until we make sure it works properly with production data

File: app/repositories/hearing_repository.rb [hunk 3]
```diff
@@ -86,22 +86,26 @@ def to_hash(hearing)
...
+      hearing_datetime = hearing_day_hash[:scheduled_for].to_datetime.change(
         hour: time["h"].to_i,
         minute: time["m"].to_i,
         offset: time["offset"]
       )
 
-      if hearing_day[:request_type] == "C"
-        create_child_co_hearing(hearing_datetime, appeal)
+      if (hearing_type || hearing_day_hash["request_type"]) == "C"
+        create_child_co_hearing(hearing_datetime, appeal, hearing_location_attrs: hearing_location_attrs)
```
[the remark this answers is not in the document]
Comment (sharonwarner) [reply app/repositories/hearing_repository.rb]: That is all correct, and that sounds good!

File: app/repositories/hearing_repository.rb (hunk 1 above)
Comment (sharonwarner) [reply app/repositories/hearing_repository.rb]: Not at this stage - the hearing repository should only be used for legacy hearings. All AMA hearings logic should be in hearing.rb.

File: client/app/queue/components/AssignHearingModal.jsx (hunk 2 above)
Comment (sharonwarner) [reply client/app/queue/components/AssignHearingModal.jsx]: Do you mind creating a ticket for this just to make sure we don't forget?

**Pull request merged**